tests/foreman/api/test_reporttemplates.py

```diff
@@ -836,3 +839,94 @@ def test_positive_installable_errata(
     installable_errata = report[0]
     assert FAKE_1_CUSTOM_PACKAGE_NAME in installable_errata['Packages']
     assert installable_errata['Erratum'] == ERRATUM_ID
+
+
+@pytest.mark.tier2
+@pytest.mark.rhel_ver_match(r'^(?!6$)\d+$')
+def test_positive_installed_products(
+    target_sat,
+    rhel_contenthost,
+    default_location,
+    function_sca_manifest_org,
+):
+    """Generate 'Host - Installed Products' report for an SCA host.
+
+    :id: d290daa2-aaba-4f4d-8eee-d8a540415320
+
+    :parametrized: yes
+
+    :setup:
+        1. RH content published in a CV, promoted to LCE, AK created.
+           All inside an SCA-enabled organization.
+        2. A RHEL content host.
+
+    :steps:
+        1. Register the content host using the AK.
+        2. Generate 'Host - Installed Products' report.
+        3. Verify the report generated from the template.
+
+    :expectedresults:
+        1. Report is generated with correct values.
+
+    :CaseImportance: Medium
+    """
+    org = function_sca_manifest_org
+    lce_name = gen_string('alpha')
+    cv_name = gen_string('alpha')
+    sys_tags = {'role': gen_string('alpha'), 'usage': gen_string('alpha')}
+
+    for key, val in sys_tags.items():
+        assert (
+            rhel_contenthost.execute(f'subscription-manager {key} --set {val}').status == 0
+        ), f'Setting of {key} failed.'
+
+    rh_repo = {
+        'basearch': DEFAULT_ARCHITECTURE,
+        'product': REPOS['rhae2']['product'],
+        'name': REPOS['rhae2']['name'],
+        'reposet': REPOS['rhae2']['reposet'],
+        'releasever': None,
+    }
+    repo_id = target_sat.api_factory.enable_sync_redhat_repo(rh_repo, org.id)
+    cv = target_sat.api_factory.cv_publish_promote(cv_name, lce_name, repo_id, org.id)
+    ak = target_sat.api.ActivationKey(
+        content_view=cv, organization=org, environment=cv.environment[-1]
```

Comment:
Just an query.. 
What if we pass `environment=lce_name` directly instead of `environment=cv.environment[-1]`?

As `CV` has been published & `CVV` has been promoted to respective `LCE`, I think for `AK` creation getting `LCE` details **explicitly** is not needed (like this way `cv.environment[-1]`) unless and untill we are going to use previously published/promoted `CV`/`CVV`

Reply:
In that case it will fail to create the AK with `Received HTTP 404 response: {"displayMessage":"Couldn't find environment 'Abc'","errors":["Couldn't find environment 'Abc'"]}`

I think we expect a `LifecycleEnvironment` object or it's id [here](https://github.com/SatelliteQE/nailgun/blob/cf4ee07b826b4f4cba0c0ce5c91c664f0a009bd7/nailgun/entities.py#L215)

Reply:
Okay, thanks for detailing the issue, this is something different error.